### ISSUE TITLE
Enhancement: [no-floating-promises] allow floating promises in some situations 

### ISSUE THREAD

**Issue by eng-dibo** [typescript-eslint/typescript-eslint#5141]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

.

### Description

in some situations it is nonsense to check if the promise is floating, especially when a function returns a Promise

examples:
```
function getPromise(): Promise<any>{
  return new Promise( r=>r('') )
}
```
```
new Promise( r=r('')).then( value=> new Promise( r=> r(value) )
```

in both cases we intend to return a floating promise, the function's consumer has to handl it

### Fail

```typescript
.
```

### Pass

```typescript
.
```

**Comment by JoshuaKGoldberg** [comment]
@eng-dibo you didn't fill out the full rule template and your description doesn't explain _why_ you want to allow floating Promises in those situations. Nor does it really explain what those situations are. I'm going to close this issue now because there isn't enough info in it for us to take action on it.

Feel free to comment back or raise a new issue if you can provide the needed information:
* _What_ are the cases the rule should ignore?
* _Why_ do you want the rule to exclude those cases?
* _How_ can those cases be distinguished from code that shouldn't be ignored?

Keep in mind you can always use an `// eslint-disable-next-line` or similar comment to disable the rule for individual parts of code. https://eslint.org/docs/user-guide/configuring/rules#disabling-rules
